### DISCUSSION TITLE
Welcome to supervision Discussions!

### DISCUSSION THREAD

**Discussion by SkalskiP** [roboflow/supervision#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vanillabrand** [comment]
The football example - I would love to know how that is done. Is there source code available?

**Reply by SkalskiP** [reply]
You can find quite old implementation here: https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-track-football-players.ipynb

**Reply by vanillabrand** [reply]
Thanks man :) Is there anything for doing similar with YOLOv8 etc? :) thanks!

**Comment by aniket-chakraborty2001** [comment]
I have some doubts regarding supervision version. 

My grounding Dino model requires supervision 0.4.0 to run (this notification is showed while installing some higher versions of supervision).

But when I am using the RoboFlow demo of Grounded SAM, it has a function MaskAnnotator for annotating the masks, I think supervision 0.4.0 is not compatible with this version (I tried to run the cell and got the error maskannotater is not defined),

Now the question is, how can install two different versions of supervision for two different work?

Advise what to do or suggest something

**Reply by onuralpszr** [reply]
@aniket-chakraborty2001 hello, first of installing 2 versions not the best idea. Also can you share exact link or source of which demo you referring to so we can check it out. Plus Installing latest should be good enough the handle Grounded SAM and others. Plus MaskAnnotator is pretty much exist in latest version as well. 

MaskAnnotator : https://supervision.roboflow.com/latest/detection/annotators/#supervision.annotators.core.MaskAnnotator

**Reply by SkalskiP** [reply]
Hi @aniket-chakraborty2001 👋🏻 Which version of Grounding DINO are you currently using? The official repository was updated some time ago and should now be compatible with the latest supervision.

I agree with @onuralpszr's opinion. Installing two versions of supervision is generally not a good idea and will likely lead to further complications.

I recommend migrating your Grounding DINO setup to use the new supervision. This will give you access to the latest features, including `MaskAnnotator`. It shouldn't be too complex.

**Comment by aniket-chakraborty2001** [comment]
I am following a youtube video link that is 1 year older the link is
https://youtu.be/sQnI_4XjfUQ?si=XWmqCWGiVLolE7q0

After attempting this process, the code does not work. The reason is:

1. *No module name GroundingDino (when we use from
groundingdino.util.inference import Model) (Google Collab)*

Since it did not work, I moved to the system based approach (vs code). I
don't have a GPU. I installed all the requirements plus the supervision
0.4.0 version. The old API (load_model, load_image, predict, annotate)
works fine for me.  https://youtu.be/xbyQy5bbdDg?si=FJLkNDiMVwznAViv. For
the second (New API, Model, Pedict_with_class) also worked for me using the
same supervision version 0.4.0

When I am going through the GroundingDino SAM notebook, it throws an error
sv.MaskAnnotator() not found, did you mean BoxAnnotator? I came to know
that the MaskAnnotator is available in higher versions of supervision.

I observe that according to the requirements.txt it needs
supervision>=0.22.0. The thing is if i installed it, it shows GroundingDino
is compatible with supervision 0.4.0 and you installed version 0.23.0

I am attaching the pictures along with the python files on which I worked.

**Reply by onuralpszr** [reply]
@aniket-chakraborty2001 👋  Hello,

First of I would recommend you to follow this notebook for handling (Grounding Dino) https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/zero-shot-object-detection-with-grounding-dino.ipynb 

Secondly If you check Grounding Dino most recent requirements.txt it does use more recent version of supervision (https://github.com/IDEA-Research/GroundingDINO/blob/main/requirements.txt) in the video person is using certain commit to handling it. 

For Grounding Dino + SAM, you can get bbox information from Grounding Dino result feed into SAM v1 but you can even use SAM v2 (better) (For basic use case of SAM2 notebook link is https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-segment-images-with-sam-2.ipynb) 

I hope those links will make your development more smoother.

**Comment by aniket-chakraborty2001** [comment]
ok, I will try the google colab notebooks you shared with me and lets see
how it goes.

I will let you know if I have any questions or concerns and also when it is
done.

**Comment by aniket-chakraborty2001** [comment]
The notebooks you shared work great for me. Moreover, I have tried the
Grounded SAM notebook available on the Roboflow github repository.
To do this, I am using the supervision version 0.6.0, as mentioned there.

*Query-1: *However, I am interested to know if I can give different colored
segmentation masks to the supervision MaskAnnotator class in the
supervision's 0.6.0 version. I read in the supervision's online
documentation that there is a parameter called custom_color_lookup and it
is set as none by default. Can I use it in supervision's version 0.6.0? And
if yes, then how?

*Query-2:* I am using a grounding dino model on an image to detect walls,
ceiling and floor. It has detected three walls, but in the detections, I
can't see any text label. Should I increase the text_threshold parameter
for this? In this same image (Result-1.png), I want to give different
colors for two different walls. That's why I asked the first query.
